src/main/scala/scorex/core/network/DeliveryTracker.scala

```diff
@@ -223,17 +299,36 @@ object DeliveryTracker {
             maxDeliveryChecks: Int,
             nvsRef: ActorRef,
             settings: ErgoSettings): DeliveryTracker = {
-    val cacheSettings = settings.cacheSettings.network
-    val bloomFilterCapacity = cacheSettings.invalidModifiersBloomFilterCapacity
-    val bloomFilterExpirationRate = cacheSettings.invalidModifiersBloomFilterExpirationRate
-    val frontCacheSize = cacheSettings.invalidModifiersCacheSize
-    val frontCacheExpiration = cacheSettings.invalidModifiersCacheExpiration
     new DeliveryTracker(
       system,
       deliveryTimeout,
       maxDeliveryChecks,
-      ExpiringApproximateCache.empty(bloomFilterCapacity, bloomFilterExpirationRate, frontCacheSize, frontCacheExpiration),
+      settings.cacheSettings.network,
+      settings.scorexSettings.network.desiredInvObjects,
       nvsRef
     )
   }
+
+  implicit class MapPimp[K, V](underlying: mutable.Map[K, V]) {
```

Comment:
This class is done in ToDownloadProcessor already (with just adjust function there), please do a common implementation.

Reply:
This is `mutable.Map` and `ToDownloadProcessor` has immutable ... I thought I would do that after rewriting `DeliveryTracker` in an immutable way.